src/main/java/com/minelittlepony/settings/ValueConfig.java

```diff
@@ -0,0 +1,41 @@
+package com.minelittlepony.settings;
+
+import com.google.gson.GsonBuilder;
+import com.mumfrey.liteloader.modconfig.AdvancedExposable;
+
+import java.io.File;
+
+/**
+ * A sensible config container that actually lets us programatically index values by a key.
+ *
+ * Reflection because Mumfrey pls.
+ *
+ */
+// Mumfrey pls.
+public class ValueConfig implements AdvancedExposable {
+
+    @Override
+    public void setupGsonSerialiser(GsonBuilder gsonBuilder) {
+        gsonBuilder.registerTypeAdapter(Value.class, new ValueSerializer());
+    }
+
+    @Override
+    public File getConfigFile(File configFile, File configFileLocation, String defaultFileName) {
+        return null;
+    }
+
+    public interface Setting extends Value<Boolean> {
```

Comment:
Do Values really need to be wrapped?

Reply:
Kinda, yeah. Your json serialization is going to result in the value<> objects in the base config class being replaced every time you load it. So we need a way to reliably get a handle on the current instance.